one-offs/images-from-boms.go

```diff
@@ -0,0 +1,187 @@
+	l.PushBack(s.Fiat.WithName("fiat"))
+	l.PushBack(s.Front50.WithName("front50"))
+	l.PushBack(s.Gate.WithName("gate"))
+	l.PushBack(s.Igor.WithName("igor"))
+	l.PushBack(s.Kayenta.WithName("kayenta"))
+	l.PushBack(s.MonitoringDaemon.WithName("monitoring-daemon"))
+	l.PushBack(s.Orca.WithName("orca"))
+	l.PushBack(s.Rosco.WithName("rosco"))
+	return l
+}
+
+type Dependencies struct {
+	Consul Service `yaml:"consul"`
+	Redis  Service `yaml:"redis"`
+	Vault  Service `yaml:"vault"`
+}
+
+type Service struct {
+	Commit  string `yaml:"commit,omitempty"`
+	Version string `yaml:"version"`
+	name    string
+}
+
+func (s *Service) WithName(n string) *Service {
+	s.name = n
+	return s
+}
+
+//type ArtifactSources struct {
+//	DebianRepository   string `yaml:"debianRepository"`
+//	DockerRegistry     string `yaml:"dockerRegistry"`
+//	GitPrefix          string `yaml:"gitPrefix"`
+//	GoogleImageProject string `yaml:"gitPrefix,omitempty"`
+//}
+
+type Image struct {
+	Digest string   `json:"digest"`
+	Tags   []string `json:"tags"`
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+	storageSvc, err := storage.NewClient(ctx, option.WithCredentialsFile(*jsonKeyPath), option.WithScopes(storage.ScopeReadOnly))
+	if err != nil {
+		log.Fatalf("Error generating new storage client: %v", err)
+	}
+
+	boms := make([]*Bom, 0, 100)
+	iter := storageSvc.Bucket(*bucket).Objects(ctx, &storage.Query{Prefix: "bom/"})
+	for obj, err := iter.Next(); err == nil; obj, err = iter.Next() {
+		if releaseRegexp.Match([]byte(obj.Name)) {
+			log.Println(obj.Name)
+			r, err := storageSvc.Bucket(*bucket).Object(obj.Name).NewReader(ctx)
+			if err != nil {
+				log.Fatalf("error reading object %v: %v", obj.Name, err)
+			}
+			bom := &Bom{}
+			if err = yaml.NewDecoder(r).Decode(bom); err != nil {
+				log.Fatalf("error decoding bom %v: %v", obj.Name, err)
+			}
+			boms = append(boms, bom)
+			log.Printf("%+v", bom)
+			//break
+		}
+	}
+
+	// Go get all the images and index them by tag
+	imagesByTag := make(map[string]*Image, 1000)
+	l := boms[0].Services.List()
+	for svcElem := l.Front(); svcElem != nil; svcElem = svcElem.Next() {
+		svc := svcElem.Value.(*Service)
+		log.Printf("Executing gcloud for %v\n", svc.name)
+		cmd := exec.Command("gcloud", "container", "images", "list-tags", fmt.Sprintf("gcr.io/%v/%v", *fromProjID, svc.name), "--format", "json")
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			log.Fatalf("error getting gcloud stdout: %v", err)
+		}
+		if err := cmd.Start(); err != nil {
+			log.Fatalf("error starting: %v", err)
+		}
+
+		images := make([]*Image, 0, 100)
+		if err := json.NewDecoder(stdout).Decode(&images); err != nil {
+			log.Fatalf("error decoding: %v", err)
+		}
+		if err := cmd.Wait(); err != nil {
+			log.Fatalf("error waiting: %v", err)
+		}
+
+		count := 0
+		for _, img := range images {
+			for _, tag := range img.Tags {
+				fullTag := fmt.Sprintf("%v:%v", svc.name, tag)
+				imagesByTag[fullTag] = img
+			}
+			count++
```

Comment:
(minor) Should this be in the inner loop for the log line below (which counts tags) to be correct?

Reply:
Added separate counters